core/src/main/java/de/danoeh/antennapod/core/feed/MediaType.java

```diff
@@ -2,19 +2,55 @@
 
 import android.text.TextUtils;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
 public enum MediaType {
-	AUDIO, VIDEO, UNKNOWN;
-
-	public static MediaType fromMimeType(String mime_type) {
-		if (TextUtils.isEmpty(mime_type)) {
-			return MediaType.UNKNOWN;
-		} else if (mime_type.startsWith("audio")) {
-			return MediaType.AUDIO;
-		} else if (mime_type.startsWith("video")) {
-			return MediaType.VIDEO;
-		} else if (mime_type.equals("application/ogg")) {
-			return MediaType.AUDIO;
-		}
-		return MediaType.UNKNOWN;
-	}
+    AUDIO, VIDEO, UNKNOWN;
+
+    private static final Set<String> AUDIO_APPLICATION_MIME_STRINGS = new HashSet<>(Arrays.asList(
+            "application/ogg",
+            "application/opus",
+            "application/x-flac"
+    ));
+
+    // based on https://developer.android.com/guide/topics/media/media-formats
+    static final Set<String> AUDIO_SUFFIXES = new HashSet<>(Arrays.asList(
+            "3gp", "aac", "amr", "flac", "imy", "m4a", "mid", "mkv", "mp3", "mp4", "mxmf", "oga",
+            "ogg", "ogx", "opus", "ota", "rtttl", "rtx", "wav", "xmf"
+    ));
+
+    static final Set<String> VIDEO_SUFFIXES = new HashSet<>(Arrays.asList(
+            "3gp", "mkv", "mp4", "ogg", "ogv", "ogx", "webm"
+    ));
+
+    public static MediaType fromMimeType(String mimeType) {
+        if (TextUtils.isEmpty(mimeType)) {
+            return MediaType.UNKNOWN;
+        } else if (mimeType.startsWith("audio")) {
+            return MediaType.AUDIO;
+        } else if (mimeType.startsWith("video")) {
+            return MediaType.VIDEO;
+        } else if (AUDIO_APPLICATION_MIME_STRINGS.contains(mimeType)) {
+            return MediaType.AUDIO;
+        }
+        return MediaType.UNKNOWN;
+    }
+
+    /**
+     * @param suffixWithoutDot the file suffix (extension) without the dot
```

Comment:
I think it would be more clear to name the method "fromFileExtension"

Reply:
Makes sense. I actually prefer file extension to suffix as well. It's more descriptive.